utils.groovy

```diff
@@ -396,4 +396,29 @@ def get_cosa_img(pipecfg, stream) {
     return utils.substituteStr(cosa_img, [STREAM: stream])
 }
 
+// Run a closure inside a context that has access to the AWS Build
+// Upload credentials.
+def withAWSBuildUploadCredentials(Closure body) {
+    withCredentials([file(variable: 'AWS_CONFIG_FILE',
+                          credentialsId: 'aws-build-upload-config')]) {
+        utils.syncCredentialsIfInRemoteSession(['AWS_CONFIG_FILE'])
+        body()
+    }
+}
+
+// Run commands inside a context that has access to the AWS Build
+// Upload credentials.
+def shwrapWithAWSBuildUploadCredentials(cmds) {
+    withAWSBuildUploadCredentials() {
+        shwrap(cmds)
+    }
+}
+
+// Return true or false based on if the AWS Build Upload credential exists
+def AWSBuildUploadCredentialExists() {
+    creds = [file(variable: 'AWS_BUILD_UPLOAD_CONFIG',
+                credentialsId: 'aws-build-upload-config')]
+    return utils.credentialsExist(creds)
```

Comment:
Minor: feels like we can inline this.

Reply:
you mean instead of using a separate `creds` variable. I had it like that before but didn't like it so I switched it to this.